nexus/capabilities/src/sled_agent.rs

```diff
@@ -0,0 +1,57 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Nexus capabilities for creating sled-agent clients
+
+use crate::Base;
+use nexus_db_queries::context::OpContext;
+use nexus_db_queries::db::lookup;
+use nexus_db_queries::db::lookup::LookupPath;
+use sled_agent_client::Client;
+use slog::o;
+use std::net::SocketAddrV6;
+use std::sync::Arc;
+use uuid::Uuid;
+
+#[async_trait::async_trait]
+pub trait SledAgent: Base {
+    /// Operational context used for sled lookups to create sled agent clients
+    fn opctx_sled_client(&self) -> &OpContext;
+
+    fn sled_lookup<'a>(
+        &'a self,
+        opctx: &'a OpContext,
+        sled_id: Uuid,
+    ) -> lookup::Sled<'a> {
+        LookupPath::new(opctx, self.datastore()).sled_id(sled_id)
+    }
+
+    fn sled_client(&self, id: Uuid, address: SocketAddrV6) -> Client {
+        let log = self.log().new(o!("SledAgent" => id.to_string()));
+        let dur = std::time::Duration::from_secs(60);
+        let client = reqwest::ClientBuilder::new()
+            .connect_timeout(dur)
+            .timeout(dur)
+            .build()
+            .unwrap();
+        Client::new_with_client(&format!("http://{address}"), client, log)
+    }
```

Comment:
> Do we ever really want to expose this?

I think so; this is the function that `blueprint-execution` calls, for example, because it already knows the sled-agent address and doesn't need to look it up again.

> If not, where would we put this functionality? Free functions in this module?

Yep; private functions in this module would be perfectly reasonable.